packages/binding-http/src/credential.ts

```diff
@@ -47,6 +50,11 @@ export class BasicCredential extends Credential {
             "authorization",
             "Basic " + Buffer.from(this.username + ":" + this.password).toString("base64")
         );
+        let headerName = "authorization";
+        if (this.options !== undefined && this.options.in === "header" && this.options.name !== undefined) {
+            headerName = this.options.name;
+        }
+        result.headers.set(headerName, "Basic " + Buffer.from(this.username + ":" + this.password).toString("base64"));
```

Comment:
Here we are setting two headers (`authorization` + `headerName`), but the logic should be that if `options.name` is defined it should be used in place of `authorization`.